File: 3-tier-application/applicationend/main.java
package com.example.demo;

import org.springframework.boot.*;
import org.springframework.boot.autoconfigure.*;
import org.springframework.web.bind.annotation.*;
import java.sql.*;

@SpringBootApplication
@RestController
public class DemoApplication {

  @GetMapping("/")
  String home() {
    String result = "";
    try {
      Connection conn = DriverManager.getConnection(
        "jdbc:mysql://mysql.backend.svc.cluster.local:3306/appdb", "root", "rootpass");
      Statement stmt = conn.createStatement();
      ResultSet rs = stmt.executeQuery("SELECT * FROM users");
      while(rs.next()) {
        result += rs.getInt("id") + ": " + rs.getString("name") + "<br/>";
      }
      conn.close();
    } catch (Exception e) {
      result = "DB Error: " + e.getMessage();
    }
    return "<h1>Application Tier (Java)</h1>" + result;
  }

  public static void main(String[] args) {
    SpringApplication.run(DemoApplication.class, args);
  }
}

File: 3-tier-application/applicationend/main.py
from flask import Flask
import mysql.connector
from mysql.connector import Error

app = Flask(__name__)

@app.route("/")
def home():
    result = ""
    try:
        connection = mysql.connector.connect(
            host='mysql.backend.svc.cluster.local',
            database='appdb',
            user='root',
            password='rootpass'
        )

        if connection.is_connected():
            cursor = connection.cursor()
            cursor.execute("SELECT * FROM users")
            rows = cursor.fetchall()
            for row in rows:
                result += f"{row[0]}: {row[1]}<br/>"

    except Error as e:
        result = f"DB Error: {e}"

    finally:
        if 'cursor' in locals():
            cursor.close()
        if 'connection' in locals() and connection.is_connected():
            connection.close()

    return f"<h1>Application Tier (Python)</h1>{result}"

if __name__ == "__main__":
    app.run(host='0.0.0.0', port=5000)
